src/observer/consume.py

```diff
@@ -112,37 +112,57 @@ def single(endpoint, idfn=None, **kwargs):
     idfn = idfn or default_idfn
     return upsert(idfn(data), content_type, data)
 
-def all_items(endpoint, idfn=None, **kwargs):
+def all_items(endpoint, idfn=None, some_fn=None):
     """consumes all items from the given `endpoint` and then creates/inserts them into the database.
     items are inserted into the database in groups of 100.
-    `idfn` is used to derive the value for `models.RawJSON.json_id`."""
+    `idfn` is used to derive the value for `models.RawJSON.json_id`.
+
+    if `some_fn` then per-page consumption is broken as soon as some_fn(item) returns False.
+    whole page of results is upserted even if `some_fn` breaks part-way through results page.
```

Comment:
no, change this. If we're going to be providing a function that essentially filters results and is page-agnostic, then upsert behaviour should be consistent with it.